src/types/dict.rs

```diff
@@ -41,11 +41,11 @@ impl PyDict {
     pub fn from_sequence(py: Python, seq: PyObject) -> PyResult<&PyDict> {
         unsafe {
             let dict = py.from_owned_ptr::<PyDict>(ffi::PyDict_New());
-            match ffi::PyDict_MergeFromSeq2(dict.into_ptr(), seq.into_ptr(), 1i32) {
-                0 => Ok(dict),
-                -1 => Err(PyErr::fetch(py)),
-                _ => unreachable!(),
-            }
+            err::error_on_minusone(
+                py,
+                ffi::PyDict_MergeFromSeq2(dict.into_ptr(), seq.into_ptr(), 1i32),
```

Comment:
nit: does this have to be `i32`? Seems better to actually leave the type unspecified since it's a `c_int`.